### PR TITLE
Add regex dos entry

### PR DESCRIPTION
**Pull request by PiranhaSa**

### CODE REVIEW

**Review by BlueSquare1 — DISMISSED** [review]
Please fix folder name `REGES_DOS`

**Review by rabsondev — COMMENTED** [review]
lgtm

**Review by rabsondev — APPROVED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/meta.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+  "risk_rating": "high",
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/meta.json:2]:
```suggestion
  "risk_rating": "medium",
```

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+To mitigate Regular Expression Denial of Service (ReDoS) vulnerabilities consider the following recommendations:
+
+* If possible avoid using user supplied input when generating `regex`.
+* Validate User Input.
+* Use Regular Expression Libraries.
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md:5]: Way too poor.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md (hunk 2 above)
Comment (PiranhaSa) [reply MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md]: what about now

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+
+* **Validate User Input:**
+Implement robust validation mechanisms to ensure that user-provided input for regex construction adheres to predefined criteria. Validate the length, complexity, and structure of input patterns to mitigate the risk of maliciously crafted expressions. By validating input at the outset, you can preemptively identify and reject potentially harmful patterns.
+
+
+* **Utilize Specialized Regular Expression Libraries:**
+Leverage specialized regular expression libraries and tools that offer built-in safeguards against ReDoS vulnerabilities. These libraries often incorporate advanced optimizations and security features designed to mitigate the impact of resource-intensive regex patterns. By utilizing trusted and vetted libraries, you can enhance the resilience of your regex implementations and minimize exposure to exploitation.
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md:12]: Like which ones. Are there Regexp libraries not vulnerable to DOS?

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+
+* **Implement Rate Limiting and Timeout Mechanisms:**
+Apply rate limiting and timeout mechanisms to restrict the execution time and resource consumption associated with regex evaluation. Set appropriate limits on the complexity and duration of regex matching operations to prevent excessive computational overhead. By imposing reasonable constraints on regex processing, you can mitigate the risk of ReDoS attacks and ensure the stability of your application under varying input conditions.
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md:16]: Missing code examples.

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md (hunk 4 above)
Comment (PiranhaSa) [reply MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md]: can you check now

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+  
+      // Method to validate user input
+      public static boolean isValidInput(String input) {
+          // Define a regex pattern to match any regex metacharacters
+          String regexPattern = "[\\\\^$.*+?()\\[\\]{}|]";
+  
+          // Check if the input contains any regex metacharacters
+          return !Pattern.compile(regexPattern).matcher(input).find();
+      }
+
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md:55]: How are these better?

**Review by 3asm — CHANGES_REQUESTED** [review]

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md (hunk 5 above)
Comment (PiranhaSa) [reply MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md:55]: using validation method here isvalidinput

Comment (PiranhaSa) [reply MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/recommendation.md]: I've changed the example

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/description.md [hunk 6]
```diff
@@ -0,0 +1,98 @@
+Regular Expression Denial of Service (ReDoS) is a security vulnerability that occurs when user input is utilized to construct a regular expression. In the presence of a carefully crafted regex pattern, the application may expend a substantial amount of resources, potentially resulting in a denial of service.
+
```
Comment (oussamaessaji) [inline MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/description.md:2]: Can you mention the Evil Regexes

Comment (ErebusZ) [reply MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/description.md:2]: I think that would be specific to each case

File: MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/description.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+- ([a-zA-Z]+)*
+
+
+- (a|aa)+
+
+
+- (a|a?)+
+
+
+- (.*a){x} for x > 10
```
Comment (3asm) [inline MOBILE_CLIENT/COMMON/_HIGH/REGEX_DOS/description.md:17]:
```suggestion
- `(a+)+`
- `([a-zA-Z]+)*`
- `(a|aa)+`
- `(a|a?)+`
- `(.*a){x}` for x > 10
```

**Review by 3asm — APPROVED** [review]

**Pull request merged**